ci/check-each-commit.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+if [ "$1" = "" ]; then
+	echo "USAGE: $0 remote/head_branch"
+	echo "eg $0 upstream/main"
+	exit 1
+fi
+
+set -e
+set -x
+
+if [ "$(git log --pretty="%H %D" | grep "^[0-9a-f]*.* $1")" = "" ]; then
+	echo "It seems like the current checked-out commit is not based on $1"
+	exit 1
+fi
+for COMMITHASH in `git log --reverse --format=format:%H $1...HEAD`; do
+	git checkout $COMMITHASH
+	cargo check
+	cd fuzz && cargo check --features=stdin_fuzz && cd ..
```

Comment:
Found a bug when tracking down a failure this caused on one of my branches.

I think this suffers from pipeline failure masking. `set -e` will only fail if the last item in a pipeline fails.

On failure, `cargo check --features=stdin_fuzz` fails, but it isn't the last item in the pipeline so it goes through the loop again (but now in the fuzz directory) where `cargo check` is run and fails. Since that command isn't a pipeline, `set -e` fails, but it has progressed the checkout so now it is ambiguous where the real failure was.

```
Previous HEAD position was ae34dea fix: Conduit::Decryptor should not panic on decryption failure
HEAD is now at 178a92f test: Update handshake fuzz test to cover more cases
+ cargo check
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
+ cd fuzz
+ cargo check --features=stdin_fuzz
    Checking lightning-fuzz v0.0.1 (/home/julian/rust-lightning/fuzz)
warning: unnecessary parentheses around assigned value
   --> src/peer_crypt.rs:250:25
    |
250 |     used_generated_data |= (is_bad_data | did_add_garbage);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses
    |
    = note: `#[warn(unused_parens)]` on by default

warning: unnecessary parentheses around assigned value
   --> src/peer_crypt.rs:288:25
    |
288 |     used_generated_data |= (is_bad_data | did_add_garbage);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses

error[E0599]: no method named `unwrap` found for struct `std::vec::Vec<u8>` in the current scope
   --> src/peer_crypt.rs:144:72
    |
144 |         assert_eq!(sender_unencrypted_msg, receiver_unencrypted_msg.unwrap().unwrap().as_slice());
    |                                                                              ^^^^^^ method not found in `std::vec::Vec<u8>`

error: aborting due to previous error; 2 warnings emitted

For more information about this error, try `rustc --explain E0599`.
error: could not compile `lightning-fuzz`.

To learn more, run the command again with --verbose.
+ for COMMITHASH in `git log --reverse --format=format:%H $1...HEAD`
+ git checkout 9f449a2f68f6cd9e6ea58f05ab807d192f6894ca
Previous HEAD position was 178a92f test: Update handshake fuzz test to cover more cases
HEAD is now at 9f449a2 review: Split outbound initialization into new()/set_up()
+ cargo check
    Checking lightning v0.0.11 (/home/julian/rust-lightning/lightning)
    Checking lightning-fuzz v0.0.1 (/home/julian/rust-lightning/fuzz)
warning: unnecessary parentheses around assigned value
   --> src/peer_crypt.rs:251:25
    |
251 |     used_generated_data |= (is_bad_data | did_add_garbage);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses
    |
    = note: `#[warn(unused_parens)]` on by default

warning: unnecessary parentheses around assigned value
   --> src/peer_crypt.rs:289:25
    |
289 |     used_generated_data |= (is_bad_data | did_add_garbage);
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these parentheses

error[E0599]: no method named `unwrap` found for struct `std::vec::Vec<u8>` in the current scope
   --> src/peer_crypt.rs:145:72
    |
145 |         assert_eq!(sender_unencrypted_msg, receiver_unencrypted_msg.unwrap().unwrap().as_slice());
    |                                                                              ^^^^^^ method not found in `std::vec::Vec<u8>`

error: aborting due to previous error; 2 warnings emitted

For more information about this error, try `rustc --explain E0599`.
error: could not compile `lightning-fuzz`.

```

Reply:
Heh, well that got rewritten anyway....test the current version?